ext/ffi/00_ffi.js

```diff
@@ -326,6 +334,33 @@
             },
           );
         }
+
+        if (needsUnpacking && !isNonBlocking) {
+          const call = this.symbols[symbol];
+          const parameters = symbols[symbol].parameters;
+          const vi = new Int32Array(2);
+          const vui = new Uint32Array(vi.buffer);
+          const b = new BigInt64Array(vi.buffer);
+
+          const params = parameters.map((_, index) => `p${index}`);
+          // Make sure V8 has no excuse to not optimize this function.
+          this.symbols[symbol] = new Function(
+            "vi",
+            "vui",
+            "b",
+            "call",
+            `return function (${params.join(", ")}) {
+            call(${params.join(", ")}${parameters.length > 0 ? ", " : ""}vi);
+            ${
+              isI64(resultType)
+                ? `const n1 = Number(b[0])`
+                : `const n1 = vui[0] + 2 ** 32 * vui[1]` // Faster path for u64
```

Comment:
Why is `BigUint64Array` avoided?

Reply:
yes it is ~8ns slower on my machine.

> has a similar attempt been done for i64?

yes but there was no perf benefit compared to BigUint64Array. Any suggestions?

Reply:
> yes but there was no perf benefit compared to BigUint64Array. Any suggestions?

I'd recommend measuring the perf of unconditionally using `BigUint64Array`, but conditionally applying [`BigInt.asIntN(64, x)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asIntN) to the extracted u64 `BigInt`. V8 optimises `BigInt.asIntN(64, x)`/`BigInt.asUintN(64, x)` to native 64-bit integers, but I'm unaware of whether it optimises handling of 64-bit input.

If you find that it yields no gain, discard the suggestion and resolve this.